src/Auth/EmailVerification/EmailVerificationRepository.php

```diff
@@ -0,0 +1,206 @@
+<?php
+namespace Nodes\Api\Auth\EmailVerification;
+
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Model as IlluminateModel;
+use Illuminate\Support\Facades\Mail;
+use Nodes\Api\Auth\Exceptions\ResetPasswordNoUserException;
+use Nodes\Database\Eloquent\Repository as NodesRepository;
+use Nodes\Api\Auth\Exceptions\MissingUserModelException;
+
+
+/**
+ * Class ResetPasswordRepository
+ *
+ * @package Nodes\Api\Auth\ResetPassword
+ */
+class EmailVerificationRepository extends NodesRepository
+{
+    /**
+     * API auth model
+     *
+     * @var \Illuminate\Database\Eloquent\Model
+     */
+    protected $authModel;
+
+    /**
+     * Constructor
+     *
+     * @author Paulius Navickas <pana@nodes.dk>
+     *
+     * @access public
+     * @param  \Nodes\Api\Auth\EmailVerification\EmailVerificationModel $model
+     * @throws \Nodes\Api\Auth\Exceptions\MissingUserModelException
+     */
+    public function __construct(EmailVerificationModel $model)
+    {
+        $this->setupRepository($model);
+
+        // Retrieve namespace of API auth model
+        $userModel = config('nodes.api.auth.model', null);
+        if (empty($userModel)) {
+            throw new MissingUserModelException('No API auth model has been specified');
+        }
+
+        // Instantiate auth model
+        $this->authModel = app($userModel);
+    }
+
+    /**
+     * Retrieve by token
+     *
+     * @author Paulius Navickas <pana@nodes.dk>
+     *
+     * @access public
+     * @param  string $token
+     * @return \Nodes\Api\Auth\EmailVerification\Model
+     */
+    public function getByToken($token)
```

Comment:
Since your field in your model is called `verification` instead of `token` this should be renamed to `getByVerification`.